spec/facility_spec.rb

```diff
@@ -23,4 +23,276 @@
+
+      expect(facility_1.registered_vehicles).to eq([cruz, camaro, bolt])
+      expect(facility_1.collected_fees).to eq(325)
+      expect(facility_2.registered_vehicles).to eq([])
+      expect(facility_2.services).to eq([])
+      expect(facility_2.register_vehicle(bolt)).to eq(nil)
+      expect(facility_2.registered_vehicles).to eq([])
+      expect(facility_2.collected_fees).to eq(0)
+    end
+  end
+
+  describe 'driver license' do
+    it 'can have a driver license' do
+      registrant_1 = Registrant.new('Bruce', 18, true )
+      registrant_2 = Registrant.new('Penny', 16 )
+      registrant_3 = Registrant.new('Tucker', 15 )
+      facility_1 = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      facility_2 = Facility.new({name: 'DMV Northeast Branch', address: '4685 Peoria Street Suite 101 Denver CO 80239', phone: '(720) 865-4600'})
+
+      expect(registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      expect(registrant_1.permit?).to eq(true)
+      expect(facility_1.administer_written_test(registrant_1)).to eq(false)
+      expect(registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      expect(facility_1.add_service('Written Test')).to eq(["Written Test"])
+      expect(facility_1.administer_written_test(registrant_1)).to eq(true)
+      expect(registrant_1.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+      expect(registrant_2.age).to eq(16)
+      expect(registrant_2.permit?).to eq(false)
+      expect(facility_1.administer_written_test(registrant_2)).to eq(false)
+
+      registrant_2.earn_permit
+      expect(facility_1.administer_written_test(registrant_2)).to eq(true)
+      expect(registrant_2.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+      expect(registrant_3.age).to eq(15)
+      expect(registrant_3.permit?).to eq(false)
+      expect(facility_1.administer_written_test(registrant_3)).to eq(false)
+
+      registrant_3.earn_permit
+      expect(facility_1.administer_written_test(registrant_3)).to eq(false)
+      expect(registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end
+
+    it 'can have a road test' do
+      registrant_1 = Registrant.new('Bruce', 18, true )
+      registrant_2 = Registrant.new('Penny', 16 )
+      registrant_3 = Registrant.new('Tucker', 15 )
+      facility_1 = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      facility_2 = Facility.new({name: 'DMV Northeast Branch', address: '4685 Peoria Street Suite 101 Denver CO 80239', phone: '(720) 865-4600'})
+
+      registrant_1.license_data
+      registrant_1.permit?
+      facility_1.administer_written_test(registrant_1)
+      registrant_1.license_data
+      facility_1.add_service('Written Test')
```

Comment:
Similar to my above comment: all these tests should include a scenario where the relevant service is not available at the facility.